File: leet-code/1047.删除字符串中的所有相邻重复项.js
/*
 * @lc app=leetcode.cn id=1047 lang=javascript
 *
 * [1047] 删除字符串中的所有相邻重复项
 */

// @lc code=start
/**
 * @param {string} s
 * @return {string}
 */
var removeDuplicates = function (s) {
  const stack = []
  for (let i = 0; i < s.length; i++) {
    const str = s[i];
    const top = stack.pop()
    if (top !== str) {
      top && stack.push(top)
      stack.push(str)
    }
  }
  return stack.join('')
};
// @lc code=end

File: leet-code-rs/src/solutions/1047_删除字符串中的所有相邻重复项.rs
/*
 * @lc app=leetcode.cn id=1047 lang=rust
 *
 * [1047] 删除字符串中的所有相邻重复项
 */

// @lc code=start

#[allow(unused)]
impl Solution {
    pub fn remove_duplicates(s: String) -> String {
        let mut st = vec![];
        for c in s.chars() {
            if !st.is_empty() && *st.last().unwrap() == c {
                st.pop();
            } else {
                st.push(c);
            }
        }

        st.iter().collect()
    }
}
// @lc code=end

#[allow(unused)]
struct Solution;

#[cfg(test)]
mod test {
    #[allow(unused)]
    use super::*;

    #[test]
    fn test_remove_duplicates() {
        //assert_eq!(Solution::remove_duplicates(vec![]),[]);
        assert!(true)
    }
}
